### DISCUSSION TITLE
Optimal solutions sensitivity analysis with CppAD and Ipopt

### DISCUSSION THREAD

**Discussion by juanmanzanero** [coin-or/CppAD#138] [Show and tell]
Hello! I just wrote a class that performs the analytical sensitivity analysis w.r.t. parameters based on optimal solutions computed from Ipopt. I have used this tool for sensitivity analysis in optimal control problems with +7000 DOFs and it works super well (it matches a numerically computed gradient using finite differences). 

![814D896A-8E37-4AEC-A964-C54A875A7441](https://user-images.githubusercontent.com/26557659/167361394-1e50da29-f40b-4126-8e5c-38576236f8b8.jpeg)

Would this be something that you want included into your repository?

The files are:

[sensitivity_analysis.h](https://github.com/juanmanzanero/lion-cpp/blob/main/lion/math/sensitivity_analysis.h)
[sensitivity_analysis.hpp](https://github.com/juanmanzanero/lion-cpp/blob/main/lion/math/sensitivity_analysis.hpp)

and a small academic test can be fount at: [sensitivity_analysis_test.cpp](https://github.com/juanmanzanero/lion-cpp/blob/main/test/math/sensitivity_analysis_test.cpp)

Cheers!

**Comment by bradbell** [comment]
If one were to add a new feature the  CppAD user API, it would have to have documentation and have automated tests. The CppAD documentation is written in omhelp, which is not widely used. I have started work on converting its documentation to sphinx, but am not yet happy with the automatic conversion. For an example of documenaiton that has been converted from omhelp to sphinx see
https://bradbell.github.io/cppad_py/doc/xsrst/cppad_py.html

Another option is to create a repository that distributes a CppAD addon; see
https://coin-or.github.io/CppAD/doc/addon.htm
Perhaps what is needed right now is for someone to lead the development of an addon repository that is easy to add to and has things like you suggest ?
